Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java

```diff
@@ -368,8 +371,154 @@ private void updateDriverStatus(Boolean status) {
         });
     }
 
+    private JSONObject getLatLng(Double lat , Double lng) throws JSONException {
+        JSONObject latLng = new JSONObject();
+        latLng.put("lat", lat);
+        latLng.put("lon", lng);
+        return latLng;
+    }
+
+    private void getRoute(Double startLat, Double startLng, Double endLat, Double endLng) {
+        StringBuilder result = new StringBuilder();
+
+        SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+
+        String regToken = sharedPref.getString("REGISTERATION_TOKEN", "null");
+        String baseUrl = sharedPref.getString("BASE_URL", "null");
+        String version = sharedPref.getString("VERSION_NAME", "null");
+        try {
+            String url = baseUrl + "/trip/route";
+            HttpURLConnection connection = (HttpURLConnection) (new URL(url).openConnection());
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "application/json");
+            connection.setRequestProperty("token", regToken);
+            connection.setRequestProperty("x-client-version", version);
+
+            JSONObject payload = new JSONObject();
+
+            JSONObject startLatLng = getLatLng(startLat, startLng);
+            JSONObject endLatLng = getLatLng(endLat, endLng);
+
+            JSONArray waypoints = new JSONArray();
+            waypoints.put(startLatLng);
+            waypoints.put(endLatLng);
+
+            payload.put("waypoints", waypoints);
+            payload.put("mode", "CAR");
+            payload.put("calcPoints", true);
+
+            OutputStream stream = connection.getOutputStream();
+            stream.write(payload.toString().getBytes());
+            connection.connect();
+            int respCode = connection.getResponseCode();
+            InputStreamReader respReader;
+            if ((respCode < 200 || respCode >= 300) && respCode != 302) {
```

Comment:
here we can have check for just if `respCode == 200`, no need to compute in case of failure. Instead just add a firebase event for failure case event: `LS_ERROR_GETTING_ROUTE`.